### PR TITLE
fix: overwrite titleTemplate with undefined

### LINKED ISSUE #169: fix: useMeta effect titleTemplate setting in nuxt.config.js doesn't work
**🐛 The bug**
The `titleTemplate`  setting in nuxt.config.js doesn't work when use useMeta in page component.

🛠️ To reproduce
The easiest way is to checkout my repo:
https://codesandbox.io/s/quirky-glitter-yg03g?file=/pages/index.vue

**🌈 Expected behaviour**
`titleTemplate`  work and don't  need to use `useMeta({ title: 'homepage', titleTemplate: '%s - my project'  })` in each page component.

### PR DESCRIPTION
**Pull request by mathe42**
fixes #169

### CODE REVIEW

**Comment by mathe42** [comment]
@danielroe you can review this.

**Comment by mathe42** [comment]
Checked other fields in vue-meta there are no other problems with this.

File: src/meta.ts [hunk 1]
```diff
@@ -38,7 +39,7 @@ export function createEmptyMeta(): MetaInfoMapper<Required<MetaInfo>> {
     __dangerouslyDisableSanitizersByTagID: {},
 
     title: undefined,
-    titleTemplate: undefined,
+    // titleTemplate: undefined,
     htmlAttrs: {},
```
Comment (danielroe) [inline src/meta.ts:43]: **question**: if we defined the custom ref here, would it remove the need for `as any`?

**Review by danielroe — APPROVED** [review]
This looks great - thanks!

**suggestion** (non-blocking): What do you think about adding an e2e test that would have failed without this addition?

File: src/meta.ts (hunk 1 above)
Comment (mathe42) [reply src/meta.ts:43]: Nope dont think that this can be done (without a complete rewrite)

**Comment by mathe42** [comment]
We could add an e2e test would be nice but I dont have the time to work on that... (in the next 1-2 Weeks...)

File: src/meta.ts [hunk 2]
```diff
@@ -109,5 +110,22 @@ export const useMeta = <T extends MetaInfo>(init?: T) => {
...
+      set(newValue) {
+        if(!_head.titleTemplate) {
+          Vue.set(_head, 'titleTemplate', newValue)
+        } else {
+          _head.titleTemplate = newValue
+        }
+        trigger()
+      }
+    }
+  })
```
Comment (danielroe) [inline src/meta.ts:128]:
```suggestion
  refs.titleTemplate = customRef((track, trigger) => {
    return {
      get() {
        track()
        return _head.titleTemplate
      },
      set(newValue) {
        if (!_head.titleTemplate) {
          Vue.set(_head, 'titleTemplate', newValue)
        } else {
          _head.titleTemplate = newValue
        }
        trigger()
      },
    }
  })
```

**Comment by danielroe** [comment]
Switching to #220 to push e2e test etc.

**Pull request closed without merging**